Jenkinsfile

```diff
@@ -0,0 +1,418 @@
+                    }
+                }
+
+                stage('SDK') {
+                    when {
+                        expression {
+                            context.isStageIncluded()
+                        }
+                    }
+                    agent {
+                        dockerfile {
+                            filename dockerFile
+                            args dockerMountArgs
+                        }
+                    }
+                    environment {
+                        BR2_BUILD_SAMPLE_DAEMON = "y"
+                        BR2_BUILD_PIKSI_INS_REF = "y"
+                    }
+                    steps {
+                        stageStart()
+                        gitPrep()
+                        crlKeyAdd()
+
+                        script {
+                            builder.make(target: "nano-image")
+                        }
+                    }
+                    post {
+                        always {
+                            cleanUp()
+                        }
+                    }
+                }
+
+                stage('Format Check') {
+                    when {
+                        expression {
+                            context.isStageIncluded()
+                        }
+                    }
+                    agent {
+                        dockerfile {
+                            filename dockerFile
+                            args dockerMountArgs
+                        }
+                    }
+                    steps {
+                        stageStart()
+                        gitPrep()
+                        crlKeyAdd()
+
+                        // Ensure that the docker tag in docker_version_tag is same as in Dockerfile.jenkins
+                        script {
+                            String dockerVersionTag = readFile(file: 'scripts/docker_version_tag').trim()
+                            sh("""grep "FROM .*:${dockerVersionTag}" scripts/Dockerfile.jenkins""")
+                            builder.make(target: "clang-format")
+                            sh('''
+                                | git --no-pager diff --name-only HEAD > /tmp/clang-format-diff
+                                | if [ -s "/tmp/clang-format-diff" ]; then
+                                |     echo "clang-format warning found"
+                                |     git --no-pager diff
+                                |     exit 1
+                                | fi
+                              '''.stripMargin())
+                        }
+                    }
+                    post {
+                        always {
+                            cleanUp()
+                        }
+                    }
+                }
+            }
+        }
+    }
+    post {
+        always {
+            // Common post-run function from ci-jenkins shared libs.
+            // Used to e.g. notify slack.
+            commonPostPipeline()
```

Comment:
Is this in the jenkins lib?  How do we handle things like conditionally uploading the sdk and the ccache?

Reply:
Yes, the "commonPostPipeline" is in the shared lib. All it does right now is postiung the result to Slack (jenkins-notifications).
I think the SDK/ccache uploads would be in the Jenkinsfile itself, not the shared lib (unless we can make it generic enough to be useful for several repos).

Reply:
Maybe we could expose individual helper functions for this?  This keeps things pretty opaque from the Jenkinsfile user's perspective, e.g. `commonlib.postSlackUpdate()`?

Reply:
Yes, good idea to make it more specific, I'll do that. I also wanted to keep a common "end of pipeline" hook so that we later could add things to the common lib (e.g. logging, cleanup, ...) without having to touch each separate Jenkinsfile.

Reply:
Pushed a change, which goes together with https://github.com/swift-nav/ci-jenkins/pull/29.

Reply:
👍